rand_distr/src/poisson.rs

```diff
@@ -115,11 +112,20 @@ impl Distribution<u64> for Poisson {
                     break;
                 }
             }
-            int_result
+            result
         }
     }
 }
 
+impl Distribution<u64> for Poisson {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> u64 {
+        let result: f64 = self.sample(rng);
+        assert!(result >= 0.);
```

Comment:
This is already guaranteed by the algorithm (see loop break condition), though the check is cheap enough relative to the algorithm that it doesn't matter.

Reply:
Maybe I should add `#[inline]`, so the check can be optimized away?

I think it makes sense to have the check to avoid undefined behavior, in case the the implementation somehow is wrong and returns negative numbers or nan.